snooty/test_parser.py

```diff
@@ -244,6 +244,65 @@ def test_admonition() -> None:
     )
 
 
+def test_toml_replacement() -> None:
+    path = ROOT_PATH.joinpath(Path("test.rst"))
+    project_config = ProjectConfig(
+        ROOT_PATH,
+        "",
+        source="./",
+        substitutions={"yaml": "*Yet Another Markup Language*"},
+    )
+    parser = rstparser.Parser(project_config, JSONVisitor)
+    substitution_nodes = {}
+    for k, v in project_config.substitutions.items():
+        node, diagnostics = parse_rst(parser, ROOT_PATH, v)
+        substitution_nodes[k] = node.ast
+
+    project_config.substitution_nodes = substitution_nodes
+    parser = rstparser.Parser(project_config, JSONVisitor)
+
+    page, diagnostics = parse_rst(
+        parser,
+        path,
+        """
+|yaml| ain't markup language.
+""",
+    )
+    assert diagnostics == []
+    check_ast_testing_string(
+        page.ast,
+        """<root>
+        <paragraph>
+        <substitution_reference>
+        <paragraph>
```

Comment:
I don't think the substituted text should be in its own paragraph; we should peel that off as well